build/source/dshare/outpt_stat.f90

```diff
@@ -0,0 +1,287 @@
+! SUMMA - Structure for Unifying Multiple Modeling Alternatives
+! Copyright (C) 2014-2015 NCAR/RAL
+!
+! This file is part of SUMMA
+!
+! For more information see: http://www.ral.ucar.edu/projects/summa
+!
+! This program is free software: you can redistribute it and/or modify
+! it under the terms of the GNU General Public License as published by
+! the Free Software Foundation, either version 3 of the License, or
+! (at your option) any later version.
+!
+! This program is distributed in the hope that it will be useful,
+! but WITHOUT ANY WARRANTY; without even the implied warranty of
+! MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+! GNU General Public License for more details.
+!
+! You should have received a copy of the GNU General Public License
+! along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+! used to manage output statistics of the model and forcing variables
+module output_stats
+USE nrtype
+implicit none
+private
+public :: allocStat
+public :: calcStats
+!public :: compileBasinStats
+contains
+
+ ! ******************************************************************************************************
+ ! public subroutine allocStat is called at beginning of simulation to allocate space for output statistics 
+ ! ******************************************************************************************************
+ subroutine allocStat(meta,stat,err,message)
+ USE nrtype
+ ! data structures
+ USE globalData,only:gru_struc          ! gru struct 
+ USE var_lookup,only:maxvarStat         ! number of different output statistics
+ USE data_types,only:var_info           ! meta type
+ USE data_types,only:gru_doubleVec,  &  ! x%gru(:)%var(:)%dat (dp)
+                     gru_hru_intVec, &  ! x%gru(:)%var(:)%dat (dp)
+                     gru_hru_doubleVec  ! x%gru(:)%hru(:)%var(:)%dat (dp)
+ implicit none
+
+ ! dummies
+ type(var_info),intent(in)  :: meta(:)  ! meta structure
+ class(*)      ,intent(out) :: stat     ! stats structure 
+ integer(i4b)  ,intent(out) :: err      ! error code
+ character(*)  ,intent(out) :: message  ! error message
+
+ ! locals
+ character(64)              :: cmessage ! error message
+ integer(i4b)               :: iVar     ! index var_info array 
+ integer(i4b)               :: iGRU     ! loop through GRUs
+ integer(i4b)               :: iHRU     ! loop through HRUs
+
+ ! initialize error control
+ err=0; message='allocStat/'
+
+ ! loop through grus
```

Comment:
Use whitespace between sections of the code in the following block to make it easier to read. See https://github.com/NCAR/summa/blob/develop/docs/howto/summa_coding_conventions.md#Whitespace